IdentityCore/src/webview/embeddedWebview/challangeHandlers/MSIDPKeyAuthHandler.m

```diff
@@ -86,4 +86,166 @@ + (BOOL)handleChallenge:(NSString *)challengeUrl
     return YES;
 }
 
++ (void)handleWwwAuthenticateHeader:(NSString *)wwwAuthHeaderValue
+                         requestUrl:(NSURL *)requestUrl
+                            context:(id<MSIDRequestContext>)context
+                  completionHandler:(void (^)(NSString *authHeader, NSError *error))completionHandler
+{
+    //pkeyauth word length=8 + 1 whitespace
+    wwwAuthHeaderValue = [wwwAuthHeaderValue substringFromIndex:[kMSIDPKeyAuthName length] + 1];
+    
+    NSDictionary *authHeaderParams = [self parseAuthHeader:wwwAuthHeaderValue];
+    
+    if (!authHeaderParams)
+    {
+        MSID_LOG_ERROR(context, @"Unparseable wwwAuthHeader received");
+        MSID_LOG_ERROR_PII(context, @"Unparseable wwwAuthHeader received %@", wwwAuthHeaderValue);
+    }
+    
+    NSError *error = nil;
+    NSString *authHeader = [MSIDPkeyAuthHelper createDeviceAuthResponse:requestUrl.absoluteString
+                                                          challengeData:authHeaderParams
+                                                                context:context
+                                                                  error:&error];
+    
+    if (completionHandler)
+    {
+        completionHandler(authHeader, error);
+    }
+}
+
+// Decodes the parameters that come in the Authorization header. We expect them in the following
+// format:
+//
+// <key>="<value>", key="<value>", key="<value>"
+// i.e. version="1.0",CertAuthorities="OU=MyOrganization,CN=MyThingy,DN=windows,DN=net",Context="context!"
+//
+// This parser is lenient on whitespace, and on the presence of enclosing quotation marks. It also
+// will allow commented out quotation marks
++ (NSDictionary *)parseAuthHeader:(NSString *)authHeader
```

Comment:
Didn’t have time to revisit this function yet, I remember last time we had some issue with the auth header if version= is not the first pair in the auth header. Need to check if it has been fixed in this function :P